src/erlcloud_ec2.erl

```diff
@@ -3281,7 +3634,101 @@ extract_cgw(Node) ->
       {customer_gateway_type, get_text("type", Node)},
       {customer_gateway_ip, get_text("ipAddress", Node)},
       {customer_gateway_bgpasn, get_text("bgpAsn", Node)},
-      {tag_set, 
+      {tag_set,
        [extract_tag_item(Item)
         || Item <- xmerl_xpath:string("tagSet/item", Node)]}
  ].
+
+-define(DEFAULT_MR, 1000).
+
+-spec describe_all(atom()) -> ok_error(proplist()).
+describe_all(DescribeFunction) ->
+    describe_all(DescribeFunction, ?DEFAULT_MR, default_config()).
+
+-spec describe_all(atom(), any()) -> ok_error(proplist()).
+describe_all(DescribeFunction, MaxResults)
+    when is_integer(MaxResults) ->
+    describe_all(DescribeFunction, MaxResults, default_config());
+describe_all(DescribeFunction, Config)
+    when is_record(Config, aws_config) ->
+    describe_all(DescribeFunction, ?DEFAULT_MR, Config);
+describe_all(DescribeFunction, Param1) ->
+    describe_all(DescribeFunction, Param1, ?DEFAULT_MR, default_config()).
+
+-spec describe_all(atom(), any(), any()) -> ok_error(proplist()).
+describe_all(DescribeFunction, MaxResults, Config)
```

Comment:
well...after reading all the clauses of this supper generic function i would agree with @ddosia that is does not belong to the library.
Therefore, sorry for waste of your time but let's remove it from the lib and move it you your application level, obviously nuking rest of `_all`.